src/components/AddItem.js

```diff
@@ -1,11 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { db } from '../lib/firebase';
+
+const initialFormState = {
+  itemName: '',
+  timeFrame: 7,
+  lastPurchased: null,
+};
 
 const AddItem = () => {
+  const token = 'user token';
+  const history = useHistory();
+  const [formData, setFormData] = useState(initialFormState);
+  const [error, setError] = useState('');
+  console.log(error);
+
+  const handleFormChange = (event) => {
+    event.preventDefault();
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!formData.itemName) {
+      setError(`${event.target.name} field cannot be empty`);
+    } else {
+      db.collection(token).add(formData);
+      history.push('/list');
+      setFormData(initialFormState);
+      setError('');
+    }
+  };
+
   return (
     <div>
       <h1>Add Item</h1>
+      <form onSubmit={handleSubmit}>
+        <h2>Add your item</h2>
+        <label>
+          Item Name
+          <input
+            type="text"
+            placeholder="add your item here"
+            name="itemName"
+            value={formData.itemName}
+            onChange={handleFormChange}
+          />
+          {error && <span>{error}</span>}
+        </label>
+
+        <fieldset className="fieldset">
+          <legend>Time Frame</legend>
+          <label htmlFor="timeFrame"> How soon will you buy this again?</label>
+          <br />
+          <input
+            type="radio"
+            id="soon"
+            name="timeFrame"
+            defaultChecked
```

Comment:
and then ... adding the attribute of checked like this to each component
`checked={radioValue === "soon"}`
`checked={radioValue === "kinda-soon"}`
`checked={radioValue === "not-soon"}`
will cause that radio button to be checked only if the `radioValue` state is set to the corresponding value

Reply:
I also noticed the radio button issue @LINDGRENBA brought up. Her solution sounds like something to try. You've got this!

Reply:
We removed the `event.preventDefault()` from the `handleFormChange()` and now the radio buttons are working as they should (able to select them).
And we added `onChange` and `checked` as attributes of the radio button inputs.